src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -688,12 +689,17 @@ def test_get_effective_shape(fortran_reader):
     child_idx += 1
     with pytest.raises(NotImplementedError) as err:
         _ = routine.children[child_idx].lhs._get_effective_shape()
-    assert "include a function call or expression" in str(err.value)
-    # Array access with expression in indices.
+    assert "include a function call or unsupported feature" in str(err.value)
+    # Array access with simple expression in indices.
     child_idx += 1
-    with pytest.raises(NotImplementedError) as err:
-        _ = routine.children[child_idx].lhs._get_effective_shape()
-    assert "include a function call or expression" in str(err.value)
+    shape = routine.children[child_idx].lhs._get_effective_shape()
+    assert shape == []
+    # Array access with expression involving indirect access in indices.
+    child_idx += 1
```

Comment:
Not all your fault, but I would find it much easier to understand if there would be a comment after each `child_idx+=1` to  indicate exactly what the statement in question is. E.g.:
```
    # Array access with expression involving indirect access in indices.
    # b(idx, 1+indices(1,1):) = 1
    child_idx += 1
```

Reply:
Good idea. (I think it was all my fault, just not in this PR :-) ). Done.

Reply:
Great, thanks!